http/design/endpoint.go

```diff
@@ -554,18 +554,20 @@ func (e *EndpointExpr) Finalize() {
 		}
 	}
 
+	// Make sure error types are user types and have a body.
+	for _, herr := range e.HTTPErrors {
+		herr.Finalize(e)
+	}
+
 	// Inherit HTTP errors from service and root
 	for _, r := range e.Service.HTTPErrors {
+		r.Finalize(e)
```

Comment:
Can you explain why this change is necessary? (maybe add a comment to the code with that explanation). Why can't the `Finalize` be called on the dups which would seem safer?

Reply:
When we dup the error expression, we also dup the HTTP response expression underneath the error (which means "dup"ing the response body and headers). When the HTTP error expression is defined at the API/service level the headers and body are not initialized when "dup"ing which results in nil pointer deference.